src/server/auth/token_service.py

```diff
@@ -0,0 +1,149 @@
+            except jwt.InvalidTokenError as e:
+                print(f"ERROR: server.token_required(): {e}")
+                return Responses.unauthorized_error()
+
+            user_uuid: str = data["uuid"]
+
+            with DatabaseProvider.handler() as handler:
+                handler().execute(QUERIES.SELECT_USER_UUID, (user_uuid,))
+                response = handler().fetchall()
+            if not handler.success:
+                return Responses.internal_database_error(handler.message)
+            user_exists: bool = response != []
+
+            if not user_exists:
+                return Responses.unauthorized_error()
+
+            # Returns the current logged-in users context to the routes
+            return fun(user_uuid, token, *args, **kwargs)
+
+        return decorated
+
+    @classmethod
+    def revoke_token(cls, token: str) -> Message:
+        """Revoke activated token.
+
+        Args:
+            token (str): token to revoke.
+
+        Returns:
+            Message: message returned by database handler.
+
+        """
+        decoded_token: dict[str, Any] = jwt.decode(token, cls._secret, algorithms=["HS256"])
+        expiry: datetime = datetime.fromtimestamp(decoded_token["exp"])
+
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.INSERT_REVOKED_TOKEN, (token, expiry))
+        return handler.message
+
+    @classmethod
+    def is_token_revoked(cls, token: str) -> bool:
+        """Check if token is revoked.
+
+        Args:
+            token (str): token to check.
+
+        Returns:
+            Message: True if token is revoked, False otherwise.
+
+        """
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.SELECT_REVOKED_TOKEN, (token, datetime.utcnow()))
+            is_revoked: bool = handler().fetchall() != []
+        if not handler.success:
+            return True
+        return is_revoked
+
+    @classmethod
+    def get_token(cls, unique_id: str, email: str) -> str:
+        """Create and return new token.
+
+        Args:
+            unique_id (str): user`s id,
+            email (str): user`s email.
+
+        Returns:
+            str: new token.
+
+        """
+        return jwt.encode(
+            {
+                "uuid": unique_id,
+                "email": email,
+                "exp": datetime.utcnow() + timedelta(minutes=cls._token_expiration_minutes),
+            },
+            cls._secret,
+        )
+
+    @classmethod
+    def refresh(cls, unique_id: str, token: str) -> Response:
+        """Refresh token.
+
+        Args:
+            unique_id (str): user`s id,
+            token (str): user`s token.
+
+        Returns:
+            Response: new token if succeed, appropriate error otherwise.
+
+        """
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.SELECT_USER_EMAIL_BY_UUID, (unique_id,))
```

Comment:
```suggestion
    def refresh(cls, user_uuid: str, token: str) -> Response:
        """Refresh token.

        Args:
            user_uuid (str): user`s id,
            token (str): user`s token.

        Returns:
            Response: new token if succeed, appropriate error otherwise.

        """
        with DatabaseProvider.handler() as handler:
            handler().execute(QUERIES.SELECT_USER_EMAIL_BY_UUID, (user_uuid,))
```